## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :x: [more...](#30ef550c-68d0-4c44-a68e-2192cee8f78e) |
| mint-gateway-nas.sh | :x: [more...](#38974e7f-ef05-43b2-9aad-035701066fcb) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#1e6ae5c5-dbff-4ad4-a482-5b36f5b76270) |
<a name="30ef550c-68d0-4c44-a68e-2192cee8f78e"></a>
### 12146-efb19f8/mint-zoned.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c2.minio.io:31975
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp d38cc766e066:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 42 seconds
(4/15) Running aws-sdk-ruby tests ... done in 4 seconds
(5/15) Running awscli tests ... done in 2 minutes and 20 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 1 minutes and 20 seconds
(8/15) Running minio-dotnet tests ... done in 49 seconds
(9/15) Running minio-go tests ... done in 1 minutes and 56 seconds
(10/15) Running minio-java tests ... done in 1 minutes and 52 seconds
(11/15) Running minio-js tests ... done in 48 seconds
(12/15) Running minio-py tests ... FAILED in 2 minutes and 37 seconds
{
  "name": "minio-py:test_thread_safe",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-853be951-5d39-48c7-9751-dfefc2a4df0a",
    "object_name": "ad01dd84-25dd-406c-ad71-10fa538c7a7a"
  },
  "message": "S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /minio-py-test-853be951-5d39-48c7-9751-dfefc2a4df0a, request_id: 1678F9B3BA9AE08E, host_id: 799b2f07-3cbd-4b27-9045-042c890f5d08, bucket_name: minio-py-test-853be951-5d39-48c7-9751-dfefc2a4df0a",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1640, in test_thread_safe\n    _CLIENT.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 676, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 409, in _execute\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 385, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /minio-py-test-853be951-5d39-48c7-9751-dfefc2a4df0a, request_id: 1678F9B3BA9AE08E, host_id: 799b2f07-3cbd-4b27-9045-042c890f5d08, bucket_name: minio-py-test-853be951-5d39-48c7-9751-dfefc2a4df0a\n",
  "duration": 5488
}
(12/15) Running s3cmd tests ... done in 18 seconds
(13/15) Running s3select tests ... done in 6 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="38974e7f-ef05-43b2-9aad-035701066fcb"></a>
### 12146-efb19f8/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:31819
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 788d2b65a48b:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 41 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... done in 2 minutes and 2 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 45 seconds
(8/15) Running minio-dotnet tests ... done in 35 seconds
(9/15) Running minio-go tests ... FAILED in 1 minutes and 3 seconds
{
  "args": {
    "bucketName": "minio-go-test-tidwoxlrju161ci0",
    "objectName": "g1jj3uffaxd1grwj1jccr5xubbeh16"
  },
  "duration": 6820,
  "error": "read tcp 172.17.0.8:34614->72.28.97.59:31819: read: connection reset by peer",
  "function": "GetObject(bucketName, objectName)",
  "message": "CopyN failed",
  "name": "minio-go: testSSES3EncryptedGetObjectReadSeekFunctional",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... done in 12 seconds
(10/15) Running minio-js tests ... done in 40 seconds
(11/15) Running minio-py tests ... done in 1 minutes and 14 seconds
(12/15) Running s3cmd tests ... done in 14 seconds
(13/15) Running s3select tests ... done in 2 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="1e6ae5c5-dbff-4ad4-a482-5b36f5b76270"></a>
### 12146-efb19f8/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:31846
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp a8e7587bba27:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 44 seconds
(4/15) Running aws-sdk-ruby tests ... done in 3 seconds
(5/15) Running awscli tests ... FAILED in 31 seconds
{
  "name": "awscli",
  "duration": 2671,
  "function": "aws --endpoint-url http://minio-c3.minio.io:31846 s3api copy-object --bucket awscli-mint-test-bucket-2140 --key datafile-1-kB-copy --copy-source awscli-mint-test-bucket-2140/datafile-1-kB\n",
  "status": "FAIL",
  "error": "Hash mismatch expected 084e1383b70fb0c51acc680fef370023, got ac57de7156d7fc25ac1a65f81fa3989b"
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 44 seconds
(7/15) Running minio-dotnet tests ... done in 35 seconds
(8/15) Running minio-go tests ... FAILED in 2 minutes and 13 seconds
{
  "args": {},
  "duration": 409,
  "error": "At least one of the pre-conditions you specified did not hold",
  "function": "CopyObjectPart(destination, source)",
  "message": "CopyObjectPart call failed",
  "name": "minio-go: testUnencryptedToSSES3CopyObjectPart",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... FAILED in 2 minutes and 41 seconds
{
  "name": "minio-java",
  "function": "copyObject()",
  "args": "[match etag]",
  "duration": 139,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = PreconditionFailed, message = At least one of the pre-conditions you specified did not hold, bucketName = minio-java-test-3n0quqb, objectName = minio-java-test-19qcdlb-copy, resource = /minio-java-test-3n0quqb/minio-java-test-19qcdlb-copy, requestId = 1678F96E77B2830F, hostId = 022edaa5-9213-4173-a347-dfe576152828)\nrequest={method=PUT, url=http://minio-c3.minio.io:31846/minio-java-test-3n0quqb/minio-java-test-19qcdlb-copy, headers=x-amz-copy-source-if-match: 71cff0a060f852067e443ad1e24ae26c-1\nx-amz-copy-source: /minio-java-test-36u3qqq/minio-java-test-19qcdlb\nHost: minio-c3.minio.io:31846\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210425T023838Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210425/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=412, headers=Accept-Ranges: bytes\nContent-Length: 418\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nETag: \"71cff0a060f852067e443ad1e24ae26c\"\nLast-Modified: Sun, 25 Apr 2021 02:38:38 GMT\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 1678F96E77B2830F\nX-Xss-Protection: 1; mode=block\nDate: Sun, 25 Apr 2021 02:38:38 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.execute(MinioClient.java:563), io.minio.MinioClient.executePut(MinioClient.java:904), io.minio.MinioClient.copyObject(MinioClient.java:1232), FunctionalTest.testCopyObjectMatchETag(FunctionalTest.java:1850), FunctionalTest.copyObject(FunctionalTest.java:2016), FunctionalTest.runObjectTests(FunctionalTest.java:3757), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(8/15) Running minio-js tests ... done in 46 seconds
(9/15) Running minio-py tests ... done in 2 minutes and 40 seconds
(10/15) Running s3cmd tests ... FAILED in 5 seconds
{
  "name": "s3cmd",
  "duration": "2821",
  "function": "test_put_object_multipart",
  "status": "FAIL",
  "error": "WARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Too many failures. Giving up on '/mint/data/datafile-65-MB'\nERROR: \nUpload of '/mint/data/datafile-65-MB' part 1 failed. Use\n  /usr/local/bin/s3cmd abortmp s3://s3cmd-test-bucket-30301/s3cmd-test-object-31736 8d22bab5-bb0b-4e69-83ed-14476446c0da\nto abort the upload, or\n  /usr/local/bin/s3cmd --upload-id 8d22bab5-bb0b-4e69-83ed-14476446c0da put ...\nto continue the upload.\nERROR: Upload of '/mint/data/datafile-65-MB' failed too many times (Last reason: )"
}
(10/15) Running s3select tests ... done in 6 seconds
(11/15) Running security tests ... done in 0 seconds

Executed 11 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/erasure-object.go
+++ b/cmd/erasure-object.go
@@ -1499,7 +1499,6 @@ func (er erasureObjects) restoreTransitionedObject(ctx context.Context, bucket s
 	}
 
 	_, err = er.CompleteMultipartUpload(ctx, bucket, object, uploadID, uploadedParts, ObjectOptions{
-		VersionID:        oi.VersionID,
 		MTime:            oi.ModTime,
 		Versioned:        globalBucketVersioningSys.Enabled(bucket),
 		VersionSuspended: globalBucketVersioningSys.Suspended(bucket),

--- a/pkg/kms/kms.go
+++ b/pkg/kms/kms.go
@@ -20,6 +20,8 @@ package kms
 import (
 	"encoding"
 	"encoding/json"
+
+	jsoniter "github.com/json-iterator/go"
 )
 
 // KMS is the generic interface that abstracts over
@@ -106,6 +108,7 @@ func (d *DEK) UnmarshalText(text []byte) error {
 		Ciphertext []byte `json:"ciphertext"`
 	}
 	var v JSON
+	var json = jsoniter.ConfigCompatibleWithStandardLibrary
 	if err := json.Unmarshal(text, &v); err != nil {
 		return err
 	}

--- a/pkg/kms/single-key.go
+++ b/pkg/kms/single-key.go
@@ -23,12 +23,12 @@ import (
 	"crypto/hmac"
 	"crypto/sha256"
 	"encoding/base64"
-	"encoding/json"
 	"errors"
 	"fmt"
 	"strconv"
 	"strings"
 
+	jsoniter "github.com/json-iterator/go"
 	"github.com/secure-io/sio-go/sioutil"
 	"golang.org/x/crypto/chacha20"
 	"golang.org/x/crypto/chacha20poly1305"
@@ -152,6 +152,7 @@ func (kms secretKey) GenerateKey(keyID string, context Context) (DEK, error) {
 	associatedData, _ := context.MarshalText()
 	ciphertext := aead.Seal(nil, nonce, plaintext, associatedData)
 
+	var json = jsoniter.ConfigCompatibleWithStandardLibrary
 	ciphertext, err = json.Marshal(encryptedKey{
 		Algorithm: algorithm,
 		IV:        iv,
@@ -174,9 +175,11 @@ func (kms secretKey) DecryptKey(keyID string, ciphertext []byte, context Context
 	}
 
 	var encryptedKey encryptedKey
+	var json = jsoniter.ConfigCompatibleWithStandardLibrary
 	if err := json.Unmarshal(ciphertext, &encryptedKey); err != nil {
 		return nil, err
 	}
+
 	if n := len(encryptedKey.IV); n != 16 {
 		return nil, fmt.Errorf("kms: invalid iv size")
 	}

